src/qiskit_iqm/fake_backends/iqm_fake_backend.py

```diff
@@ -0,0 +1,174 @@
+# Copyright 2022 Qiskit on IQM developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# pylint: disable=no-name-in-module,import-error,super-init-not-called,too-many-instance-attributes
+"""
+Qiskit fake backend for IQM quantum computers.
+"""
+
+from typing import Union
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import Parameter
+from qiskit.circuit.library import CZGate, Measure, RGate
+from qiskit.providers import BackendV2, JobV1
+from qiskit.transpiler import Target
+from qiskit.transpiler.coupling import CouplingMap
+from qiskit_aer import AerSimulator
+from qiskit_aer.noise import NoiseModel
+from qiskit_aer.noise.errors import depolarizing_error, thermal_relaxation_error
+
+from qiskit_iqm.fake_backends.chip_samples.example_sample import adonis_chip_sample
+from qiskit_iqm.iqm_backend import IQMBackend
+
+
+class IQMFakeBackend(IQMBackend, BackendV2):
```

Comment:
`IQMBackend` already inherits from `BackendV2`

Reply:
That's true, but we do not need to have the IQMBackend init called as there is additional code executed that is irrelevant for the fake backend. However, we want to have the relevant methods like `serialize_circuit`  also available for fake backend objects.

Suggestion would be to use `super(IQMBackend, self).__init__(self, **kwargs)` instead. Will be updated

Reply:
Yes, but we shall figure out another way to circumvent this issue - this double inheritance is not nice. If we manage to re-use the IQMBackend implementation, there will be couple of benefits:
- you will get target implementation for free and do not need to duplicate code in fake backend
- you will reuse the quantum architecture definition from IQM client instead of inventing a new data structure for the same thing.
- Less and nicer code

I will consider some alternatives and propose a solution

Reply:
This was resolved by my commits